### ISSUE TITLE
epub/mobi/html5 version

### ISSUE THREAD

**Issue by hmemcpy** [hmemcpy/milewski-ctfp-pdf#46]

**Comment by adishavit** [comment]
Yes please - I'd love an epub/mobi version too.

**Comment by latentgenius** [comment]
I'll have a go at this.

**Comment by mb21** [comment]
I'll share how far I've come in my attempt.
TL;DR: ePub successfully generated, but still a few issues.

Use pandoc 2.0 (currently not yet released, so you'll have to use the [pandoc nightly](https://github.com/pandoc-extras/pandoc-nightly) or compile it yourself.)

Then `cd milewski-ctfp-pdf/src` and:

    pandoc ctfp.tex -o category-theory-for-programmers.epub --webtex -M author="Bartosz Milewski" -M title="Category Theory for Programmers" --resource-path=.:content/0.0:content/1.1:content/1.10:content/1.2:content/1.3:content/1.4:content/1.5:content/1.6:content/1.7:content/1.8:content/1.9:content/2.1:content/2.2:content/2.3:content/2.4:content/2.5:content/2.6:content/3.1:content/3.10:content/3.11:content/3.12:content/3.13:content/3.14:content/3.15:content/3.2:content/3.3:content/3.4:content/3.5:content/3.6:content/3.7:content/3.8:content/3.9

See https://github.com/jgm/pandoc/issues/3983 for a discussion about how pandoc handles this LaTeX project, but basically:

- use `--resource-path` will all image directories becauese of the `\renewcommand*{\subfile}` in this LaTeX
- use `--webtex` to convert MathML to SVG because most eReaders don't support MathML, see https://pandoc.org/epub.html#math


A few things are still not parsed correctly by pandoc:

- `\lettrine[lhang=0.17]{A}{category}`
- `$\Set$`
- ...?

Pandoc does handle `\renewcommand` to some extent, but obviously it cannot reimplement all of TeX macro behaviours. I guess it's debatable what should be fixed in pandoc and what things are just too non-standard in this LaTeX source for pandoc to re-implement...

**Comment by hmemcpy** [comment]
Wow, this looks fantastic!
I've removed a bunch of unneeded macros already, but it seems there's more work to be done.
Thank you so much for the effort! I'll see if I can help in any way by simplifying things further...

**Comment by onlurking** [comment]
I'm currently working on the [ePub version](https://github.com/onlurking/category-theory-for-programmers), and after being finished, i will also target mobi(using [KindleGen](https://www.amazon.com/gp/feature.html?docId=1000765211)) and HTML(I can export, but I need a better styling for the HTML version).

The project is [here](https://github.com/onlurking/category-theory-for-programmers) if you guys are interested, and I'm following this [guide](https://puppet.com/blog/how-we-automated-our-ebook-builds-pandoc-and-kindlegen) mostly.

**Comment by kevinwright** [comment]
Has there been any more progress on this, and are there any specific issue that need handling?

I'm especially interested in seeing the scala version in eBook form
